packages/next/server/config.ts

```diff
@@ -57,17 +57,11 @@ const experimentalWarning = execOnce(
   }
 )
 
+// TODO-APP: the type of this is wrong. In worker.ts it's called with only a fraction of the config, not the full config object.
 export function setHttpClientAndAgentOptions(config: NextConfig) {
```

Comment:
```suggestion
export function setHttpClientAndAgentOptions(config: Partial<NextConfig>) {
```

This should fix it so the above comment can be removed